x-pack/winlogbeat/module/powershell/_meta/fields.yml

```diff
@@ -0,0 +1,136 @@
+- key: powershell
+  title: PowerShell module"
+  description: >
+    These are the event fields specific to the module for the Microsoft-Windows-PowerShell/Operational and Windows PowerShell logs.
+  release: beta
+  fields:
+
+    - name: id
+      type: keyword
+      description: Shell Id.
+      example: Microsoft Powershell
+
+    - name: pipeline_id
+      type: keyword
+      description: Pipeline id.
+      example: "1"
+
+    - name: runspace_id
+      type: keyword
+      description: Runspace id.
+      example: "4fa9074d-45ab-4e53-9195-e91981ac2bbb"
+
+    - name: sequence
+      type: long
+      description: Sequence number of the powershell execution.
+      example: 1
+
+    - name: total
+      type: long
+      description: Total number of messages in the sequence.
+      example: 10
+
+    - name: powershell.command
+      type: group
+      description: Data related to the executed command.
+      fields:
+        - name: path
+          type: keyword
+          description: Path of the executed command.
+          example: "C:\\Windows\\system32\\cmd.exe"
+
+        - name: name
+          type: keyword
+          description: Name of the executed command.
+          example: "cmd.exe"
+
+        - name: type
+          type: keyword
+          description: Type of the executed command.
+          example: Application
+
+        - name: value
+          type: text
+          description: The invoked command.
+          example: Import-LocalizedData  LocalizedData -filename ArchiveResources
+
+        - name: invocation_details
```

Comment:
I _think_ you can declare the nested fields with a line like

`- name: invocation_details.related_command`